spring-integration-ftp/src/test/java/org/springframework/integration/ftp/inbound/FtpStreamingMessageSourceTests.java

```diff
@@ -0,0 +1,116 @@
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.annotation.InboundChannelAdapter;
+import org.springframework.integration.annotation.Transformer;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.core.MessageSource;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.ftp.FtpTestSupport;
+import org.springframework.integration.ftp.session.FtpRemoteFileTemplate;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.integration.transformer.StreamTransformer;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.scheduling.support.PeriodicTrigger;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * @author Gary Russell
+ * @since 4.3
+ *
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext
+public class FtpStreamingMessageSourceTests extends FtpTestSupport {
+
+	@Autowired
+	public PollableChannel data;
+
+	@SuppressWarnings("unchecked")
+	@Test
+	public void testAllContents() {
+		Message<byte[]> received = (Message<byte[]>) this.data.receive(10000);
+		assertNotNull(received);
+		assertThat(new String(received.getPayload()), equalTo("source1"));
+		received = (Message<byte[]>) this.data.receive(10000);
+		assertNotNull(received);
+		assertThat(new String(received.getPayload()), equalTo("source2"));
+		assertNull(this.data.receive(0));
+	}
+
+	@Configuration
+	@EnableIntegration
+	public static class Config {
+
+		@Bean
+		public QueueChannel data() {
+			return new QueueChannel();
+		}
+
+		@Bean(name = PollerMetadata.DEFAULT_POLLER)
+		public PollerMetadata defaultPoller() {
+			PollerMetadata pollerMetadata = new PollerMetadata();
+			pollerMetadata.setTrigger(new PeriodicTrigger(500));
+			pollerMetadata.setMaxMessagesPerPoll(2000);
+			return pollerMetadata;
+		}
+
+		@Bean
+		@InboundChannelAdapter(channel = "stream")
+		public MessageSource<InputStream> ftpMessageSource() {
+			FtpStreamingMessageSource messageSource = new FtpStreamingMessageSource(template(), null);
+			messageSource.setRemoteDirectory("ftpSource/");
+			return messageSource;
+		}
+
+		@Bean
+		@Transformer(inputChannel = "stream", outputChannel = "data")
+		public org.springframework.integration.transformer.Transformer transformer() {
+			return new StreamTransformer();
+		}
+
+		@Bean
+		public static FtpRemoteFileTemplate template() {
+			return new FtpRemoteFileTemplate(ftpSessionFactory());
+		}
+
+		@Bean
+		public static SessionFactory<FTPFile> ftpSessionFactory() {
```

Comment:
Why these two beans are `static`?
Looks like is just overhead.
Although it isn't critical.
Never mind